test/libsolidity/semanticTests/various/transient_storage_reentrancy_lock.sol

```diff
@@ -0,0 +1,24 @@
+contract C {
+    bool transient locked;
+    modifier nonReentrant {
+        require(!locked);
+        locked = true;
+        _;
+        locked = false;
+    }
+
+    function test(address newAddress, bool reentrancy) nonReentrant public {
+        if (reentrancy)
+            reentrantCall(newAddress);
+    }
+
+    function reentrantCall(address a) public {
+        this.test(a, false);
+    }
+}
+// ====
+// EVMVersion: >=cancun
+// compileViaYul: false
+// ----
+// test(address,bool): 0x1234abcd, true -> FAILURE
```

Comment:
So what ended up being the issue here?

Reply:
It was this spot where I missed an `SLOAD` and didn't replace it by the appropriate member of the template class:

https://github.com/ethereum/solidity/blob/3566cbd2d60d40ddee3045446bd3df58573b173a/libsolidity/codegen/LValue.cpp#L246